src/components/userProfile/EditProfile.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+import { editProfileAction } from "../../actions/profileActions";
+import firebase from "../../firebase/config";
+import "../../styles/profile.scss";
+import TopAppBar from "../header";
+
+export class EditProfile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      bio: "",
+      full_name: "",
+      image: "",
+      isProcessing: false,
+      uploadDone: false,
+      progress: 0
+    };
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleBioChange = this.handleBioChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleImageUpload = this.handleImageUpload.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.upDateSuccess === true) {
+      const { history } = nextProps;
+      history.push("/profile");
+    }
+  }
+
+  componentWillMount() {
+    const { full_name, bio, image } = this.props.profile;
+    this.setState({ full_name, bio, image });
+  }
+  handleNameChange(event) {
+    this.setState({ full_name: event.target.value });
+  }
+  handleBioChange(event) {
+    this.setState({ bio: event.target.value });
+  }
+  handleImageUpload(event) {
+    this.setState({ image: event.target.files });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const data = {
+      bio: this.state.bio,
+      full_name: this.state.full_name,
```

Comment:
destructure this as well. Ensure to go through the file and destructure where were necessary

Reply:
Done